src/main/java/org/jenkins/plugins/lockableresources/LockStepExecution.java

```diff
@@ -62,10 +66,27 @@ public static void proceed(List<String> resourcenames, StepContext context, Stri
 		}
 
 		LOGGER.finest("Lock acquired on [" + resourceDescription + "] by " + r.getExternalizableId());
-		context.newBodyInvoker().
-			withCallback(new Callback(resourcenames, resourceDescription, inversePrecedence)).
-			withDisplayName(null).
-			start();
+		try {
+			BodyInvoker bodyInvoker = context.newBodyInvoker().
+				withCallback(new Callback(resourcenames, resourceDescription, inversePrecedence)).
+				withDisplayName(null);
+			if(variable != null && variable.length()>0)
+				// set the variable for the duration of the block
+				bodyInvoker.withContext(EnvironmentExpander.merge(context.get(EnvironmentExpander.class), new EnvironmentExpander() {
+					@Override
+					public void expand(EnvVars env) throws IOException, InterruptedException {
+						final String resources = resourcenames.toString();
```

Comment:
Iterating the list would be both efficient and consistent than this. List#toString doesn't even define any kind of contract.

Reply:
You're right, I was too lazy...